### DISCUSSION TITLE
unable to deploy  a sample express webapp

### DISCUSSION THREAD

**Discussion by anildugar** [vercel/vercel#4402] [Help] [answered]
Hi 

i am unable to deploy a sample express based app, there is no error reported on now platform, but the build just doesnt finish, it keeps running on and on and on.

doest it mean new vercel doesnt support express based server apps ?

here is my source on github repo : https://github.com/anildugar/now-server

Thanks in advance

**Comment by ghost** [comment]
According to their documentation https://nextjs.org/docs/deployment, they suggest if custom servers are being implemented then we need to set up our own hosting probably AWS, Azure, Digital Ocean to name a few. I have raised a ticket and waiting for a response for their team officially.

**Comment by mcsdevv** [comment, marked as answer]
Hello @anildugar, the Vercel platform is intended for static frontends, backed by Serverless Functions where required.

Whilst technically possible to deploy an Express server as a Serverless Function, it is certainly not recommended due to the fundamentally different models of the two technologies.
